common/djangoapps/edxmako/shortcuts.py

```diff
@@ -52,6 +52,9 @@ def marketing_link(name):
         # special case for when we only want the root marketing URL
         if name == 'ROOT':
             return marketing_urls.get('ROOT')
+        # special case for new enterprise marketing url with custom tracking query params
+        if name == 'ENTERPRISE' and settings.ENTERPRISE_MARKETING_FOOTER_QUERY_PARAMS:
```

Comment:
My intention here was to leave backwards compatibility for those who do not set `ENTERPRISE_MARKETING_FOOTER_QUERY_PARAMS` to anything. Specifically, if someone were to pull down the code and change nothing, the flow of this logic would result in the business marketing url being `localhost:18000/enterprise`.

If you do add `ENTERPRISE_MARKETING_FOOTER_QUERY_PARAMS`, it's presupposed that you'll update the enterprise marketing based url as well from `localhost:18000` to in our case `business.edx.org`

I'm open to recommendation if there's a better way to approach this

Reply:
It looks like the functionality here is the same as for 'ROOT' now

Perhaps we could just detect when the marketing_urls.get(name) is a relative url
do: 
urljoin(marketing_urls.get('ROOT'), marketing_urls.get(name))
otherwise do:
marketing_urls.get(name)

then I think your PRs could be deployed independently(and thus rolled back much more reasonably)

Reply:
I will give this a shot. Thanks for the recommendation